### PR TITLE
[BUG] Fix Memory Leak in SchemaChange And Fix some DCHECK error

### LINKED ISSUE #5466: [BUG]  BE Crash in spill node and Hash Join Node
**Describe the bug**
crash with ASAN

**To Reproduce**
```
SELECT 449 UNION SELECT 670 EXCEPT SELECT 449;
```
Hash Join Node or Except Node 's child has a spill node

**Expected behavior**
BE shouldn't core

**Screenshots**
```
==50097==ERROR: AddressSanitizer: use-after-poison on address 0x7fe1cbd17000 at pc 0x000001b13dc9 bp 0x7fe264690a90 sp 0x7fe264690a88
READ of size 1 at 0x7fe1cbd17000 thread T198 (FragmentMgrThre)
    #0 0x1b13dc8 in doris::Tuple::is_null(doris::NullIndicatorOffset const&) const ../src/runtime/tuple.h:134
    #1 0x1b14189 in doris::SlotRef::get_value(doris::Expr*, doris::TupleRow*) ../src/exprs/slot_ref.h:91
    #2 0x1b1431c in doris::ExprContext::get_value(doris::TupleRow*) ../src/exprs/expr_context.h:213
    #3 0x33e0864 in doris::HashTable::equals(doris::TupleRow*) ../src/exec/hash_table.cpp:164
    #4 0x33b45ef in doris::HashTable::find(doris::TupleRow*, bool) ../src/exec/hash_table.hpp:39
    #5 0x362bebc in doris::ExceptNode::open(doris::RuntimeState*) ../src/exec/except_node.cpp:84
    #6 0x27a7c0a in doris::PlanFragmentExecutor::open_internal() ../src/runtime/plan_fragment_executor.cpp:270
    #7 0x27a7581 in doris::PlanFragmentExecutor::open() ../src/runtime/plan_fragment_executor.cpp:253
    #8 0x25fce6b in doris::FragmentExecState::execute() ../src/runtime/fragment_mgr.cpp:219
    #9 0x2602bec in doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>) ../src/runtime/fragment_mgr.cpp:423
    #10 0x262d620 in void std::__invoke_impl<void, void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&>(std::__invoke_memfun_deref, void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&) (/ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/core/target/be/lib/palo_be+0x262d620)
    #11 0x262b5d5 in std::__invoke_result<void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&>::type std::__invoke<void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&>(void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/invoke.h:95
    #12 0x2627cc1 in _ZNSt12_Bind_resultIvFMN5doris11FragmentMgrEFvSt10shared_ptrINS0_17FragmentExecStateEESt8functionIFvPNS0_20PlanFragmentExecutorEEEEPS1_S4_S9_EE6__callIvJEJLm0ELm1ELm2EEEENSt9enable_ifIXtlSt7is_voidIT_EEEvE4typeEOSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/functional:632
    #13 0x26216a2 in void std::_Bind_result<void, void (doris::FragmentMgr::*(doris::FragmentMgr*, std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>))(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>)>::operator()<>() /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/functional:718
    #14 0x261b544 in std::_Function_handler<void (), std::_Bind_result<void, void (doris::FragmentMgr::*(doris::FragmentMgr*, std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>))(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>)> >::_M_invoke(std::_Any_data const&) (/ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/core/target/be/lib/palo_be+0x261b544)
    #15 0x2734313 in std::function<void ()>::operator()() const /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/std_function.h:706
    #16 0x2aeaf89 in doris::FunctionRunnable::run() ../src/util/threadpool.cpp:42
    #17 0x2ae7f0f in doris::ThreadPool::dispatch_thread() ../src/util/threadpool.cpp:548
    #18 0x2b018d1 in void std::__invoke_impl<void, void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(std::__invoke_memfun_deref, void (doris::ThreadPool::*&)(), doris::ThreadPool*&) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/invoke.h:73
    #19 0x2b00296 in std::__invoke_result<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>::type std::__invoke<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(void (doris::ThreadPool::*&)(), doris::ThreadPool*&) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/invoke.h:95
    #20 0x2afe7f4 in void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/functional:467
    #21 0x2afc2ce in void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::operator()<, void>() /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/functional:551
    #22 0x2af7a2d in std::_Function_handler<void (), std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()> >::_M_invoke(std::_Any_data const&) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/std_function.h:316
    #23 0x2734313 in std::function<void ()>::operator()() const /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/std_function.h:706
    #24 0x2acc73a in doris::Thread::supervise_thread(void*) ../src/util/thread.cpp:385
    #25 0x7fe2c53ed1c2 in start_thread (/opt/compiler/gcc-4.8.2/lib64/libpthread.so.0+0x81c2)
    #26 0x7fe2c4a1012c in __clone (/opt/compiler/gcc-4.8.2/lib64/libc.so.6+0xe712c)

0x7fe1cbd17000 is located 0 bytes inside of 2097152-byte region [0x7fe1cbd17000,0x7fe1cbf17000)
allocated by thread T198 (FragmentMgrThre) here:
    #0 0x19fc798 in __interceptor_posix_memalign ../../../../gcc-7.3.0/libsanitizer/asan/asan_malloc_linux.cc:134
0x7fe1cbd17000 is located 0 bytes inside of 2097152-byte region [0x7fe1cbd17000,0x7fe1cbf17000)
allocated by thread T198 (FragmentMgrThre) here:
    #0 0x19fc798 in __interceptor_posix_memalign ../../../../gcc-7.3.0/libsanitizer/asan/asan_malloc_linux.cc:134
    #1 0x287588e in doris::SystemAllocator::AllocateViaMalloc(long, unsigned char**) ../src/runtime/bufferpool/system_allocator.cc:119
    #2 0x2874c07 in doris::SystemAllocator::Allocate(long, doris::BufferPool::BufferHandle*) ../src/runtime/bufferpool/system_allocator.cc:65
    #3 0x285f576 in doris::BufferPool::BufferAllocator::AllocateInternal(long, doris::BufferPool::BufferHandle*) ../src/runtime/bufferpool/buffer_allocator.cc:302
    #4 0x285dcae in doris::BufferPool::BufferAllocator::Allocate(doris::BufferPool::ClientHandle*, long, doris::BufferPool::BufferHandle*) ../src/runtime/bufferpool/buffer_allocator.cc:223
    #5 0x26fc213 in doris::BufferPool::AllocateBuffer(doris::BufferPool::ClientHandle*, long, doris::BufferPool::BufferHandle*) ../src/runtime/bufferpool/buffer_pool.cc:230
    #6 0x26f9d27 in doris::BufferPool::CreatePage(doris::BufferPool::ClientHandle*, long, doris::BufferPool::PageHandle*, doris::BufferPool::BufferHandle const**) ../src/runtime/bufferpool/buffer_pool.cc:144
    #7 0x3a9c1c1 in doris::BufferedTupleStream3::NewWritePage(long) ../src/runtime/buffered_tuple_stream3.cc:369
    #8 0x3a9daac in doris::BufferedTupleStream3::AdvanceWritePage(long, bool*) ../src/runtime/buffered_tuple_stream3.cc:448
    #9 0x3aa3f0c in doris::BufferedTupleStream3::AddRowCustomBeginSlow(long, doris::Status*) ../src/runtime/buffered_tuple_stream3.cc:911
    #10 0x36516a9 in doris::BufferedTupleStream3::AddRowCustomBegin(long, doris::Status*) ../src/runtime/buffered_tuple_stream3.inline.h:37
    #11 0x36516a9 in doris::PartitionedAggregationNode::ConstructIntermediateTuple(std::vector<doris::NewAggFnEvaluator*, std::allocator<doris::NewAggFnEvaluator*> > const&, doris::BufferedTupleStream3*, doris::Status*) ../src/exec/partitioned_aggregation_node.cc:933
    #12 0x36785fa in doris::Status doris::PartitionedAggregationNode::AddIntermediateTuple<false>(doris::PartitionedAggregationNode::Partition*, doris::TupleRow*, unsigned int, doris::PartitionedHashTable::Iterator) ../src/exec/partitioned_aggregation_node_ir.cc:142
    #13 0x3676671 in doris::Status doris::PartitionedAggregationNode::ProcessRow<false>(doris::TupleRow*, doris::PartitionedHashTableCtx*) ../src/exec/partitioned_aggregation_node_ir.cc:133
    #14 0x3672a80 in doris::Status doris::PartitionedAggregationNode::ProcessBatch<false>(doris::RowBatch*, doris::PartitionedHashTableCtx*) ../src/exec/partitioned_aggregation_node_ir.cc:55
    #15 0x3644b8b in doris::PartitionedAggregationNode::open(doris::RuntimeState*) ../src/exec/partitioned_aggregation_node.cc:325
    #16 0x3707a10 in doris::SetOperationNode::open(doris::RuntimeState*) ../src/exec/set_operation_node.cpp:147
    #17 0x362a5b7 in doris::ExceptNode::open(doris::RuntimeState*) ../src/exec/except_node.cpp:42
    #18 0x27a7c0a in doris::PlanFragmentExecutor::open_internal() ../src/runtime/plan_fragment_executor.cpp:270
    #19 0x27a7581 in doris::PlanFragmentExecutor::open() ../src/runtime/plan_fragment_executor.cpp:253
    #20 0x25fce6b in doris::FragmentExecState::execute() ../src/runtime/fragment_mgr.cpp:219
    #21 0x2602bec in doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>) ../src/runtime/fragment_mgr.cpp:423
    #22 0x262d620 in void std::__invoke_impl<void, void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&>(std::__invoke_memfun_deref, void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&) (/ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/core/target/be/lib/palo_be+0x262d620)
    #23 0x262b5d5 in std::__invoke_result<void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&>::type std::__invoke<void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&>(void (doris::FragmentMgr::*&)(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>), doris::FragmentMgr*&, std::shared_ptr<doris::FragmentExecState>&, std::function<void (doris::PlanFragmentExecutor*)>&) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/invoke.h:95
    #24 0x2627cc1 in _ZNSt12_Bind_resultIvFMN5doris11FragmentMgrEFvSt10shared_ptrINS0_17FragmentExecStateEESt8functionIFvPNS0_20PlanFragmentExecutorEEEEPS1_S4_S9_EE6__callIvJEJLm0ELm1ELm2EEEENSt9enable_ifIXtlSt7is_voidIT_EEEvE4typeEOSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/functional:632
    #25 0x26216a2 in void std::_Bind_result<void, void (doris::FragmentMgr::*(doris::FragmentMgr*, std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>))(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>)>::operator()<>() /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/functional:718
    #26 0x261b544 in std::_Function_handler<void (), std::_Bind_result<void, void (doris::FragmentMgr::*(doris::FragmentMgr*, std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>))(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>)> >::_M_invoke(std::_Any_data const&) (/ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/core/target/be/lib/palo_be+0x261b544)
    #27 0x2734313 in std::function<void ()>::operator()() const /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/std_function.h:706
    #28 0x2aeaf89 in doris::FunctionRunnable::run() ../src/util/threadpool.cpp:42
    #29 0x2ae7f0f in doris::ThreadPool::dispatch_thread() ../src/util/threadpool.cpp:548
    #30 0x2b018d1 in void std::__invoke_impl<void, void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(std::__invoke_memfun_deref, void (doris::ThreadPool::*&)(), doris::ThreadPool*&) /ssd1/opt/stdpain/workspace/doris/baidu/bdg/doris/doris-toolchain/gcc730/include/c++/7.3.0/bits/invoke.h:73
```

**Additional context**
I will fix it

### LINKED ISSUE #5476: [BUG] BE Crash In Debug Mode And Some Memory Leak
**Describe the bug**
BE DCHECK Fail
some memory leak

**To Reproduce**
## crash 1:
```
select timediff('1019-02-28 00:00:00', '2019-02-28 00:00:00'),          timediff('2015-01-25 00:00:00', '2015-02-28 22:50:00'),          timediff('2015-01-25 00:00:00', '2015-01-25 00:00:00');
```

F0306 10:23:24.630371 26238 literal.cpp:144] Check failed: _type.type == TYPE_DOUBLE (21 vs. 9) TIME

## crash 2:
HLL/BitMap mem_tracker decheck fail

## crash 3:
select * from tabl1 a join table2 b on a.k2<=> b.k2 and         a.k4<=>NULL order by a.k1,b.k1
F0306 10:30:09.713904  1667 olap_utils.h:212] Check failed: false

## crash 4:
F0306 10:32:05.967895 22717 hll.cpp:167] Check failed: _hash_set.size() < HLL_EXPLICIT_INT64_NUM Number of explicit elements(160) should be less or equal than 160

## crash 5:
F0308 15:36:56.546083  4507 tablets_channel.cpp:263] Check failed: _mem_tracker->consumption() == 0 (73728 vs. 0)

## Some Leak :
```
Direct leak of 7296 byte(s) in 48 object(s) allocated from:
    #0 0x19fbf38 in __interceptor_calloc
    #1 0x3fce0ba in event_base_once
    #2 0x31fb42d in doris::EvHttpServer::stop()
    #3 0x31f9b0d in doris::EvHttpServer::~EvHttpServer()
    #4 0x28e5c31 in std::default_delete<doris::EvHttpServer>::operator()(doris::EvHttpServer*) const
    #5 0x28e5a08 in std::unique_ptr<doris::EvHttpServer, std::default_delete<doris::EvHttpServer> >::~unique_ptr()
    #6 0x28d8bcf in doris::HttpService::~HttpService()
    #7 0x1a39a3a in main
    #8 0x7fed3760ebd4 in __libc_start_main

Direct leak of 184 byte(s) in 1 object(s) allocated from:
    #0 0x19fd480 in operator new(unsigned long)
    #1 0x28da3f4 in doris::HttpService::start()
    #2 0x1a393b6 in main
    #3 0x7fed3760ebd4 in __libc_start_main
```

```
Direct leak of 24 byte(s) in 1 object(s) allocated from:
    #0 0x19fd480 in operator new(unsigned long) 
    #1 0x32156c0 in doris::MiniLoadAction::_is_streaming(doris::HttpRequest*) 
    #2 0x3214727 in doris::MiniLoadAction::on_header(doris::HttpRequest*) 
    #3 0x31fd08d in doris::EvHttpServer::on_header(evhttp_request*) 
    #4 0x31f8f8c in on_header 
    #5 0x3fdcd11 in evhttp_read_header 
    #6 0x3fdf1c5 in bufferevent_trigger_nolock_
    #7 0x3fdf1c5 in bufferevent_readcb 
    #8 0x3fcd05e in event_persist_closure 
    #9 0x3fcd05e in event_process_active_single_queue 
    #10 0x3fcd97e in event_process_active 
    #11 0x3fcd97e in event_base_loop 
    #12 0x31fa3ef in operator() 
    #13 0x31fdb4b in _M_invoke 
    #14 0x273444b in std::function<void ()>::operator()() const 
    #15 0x2aeb0c1 in doris::FunctionRunnable::run() 
    #16 0x2ae8047 in doris::ThreadPool::dispatch_thread() 
    #17 0x2b01a09 in void std::__invoke_impl<void, void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(std::__invoke_memfun_deref, void (doris::ThreadPool::*&)(), doris::ThreadPool*&) 
    #18 0x2b003ce in std::__invoke_result<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>::type std::__invoke<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(void (doris::ThreadPool::*&)(), doris::ThreadPool*&) 
    #19 0x2afe92c in void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) 
    #20 0x2afc406 in void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::operator()<, void>() 
    #21 0x2af7b65 in std::_Function_handler<void (), std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()> >::_M_invoke(std::_Any_data const&) 
    #22 0x273444b in std::function<void ()>::operator()() const 
    #23 0x2acc872 in doris::Thread::supervise_thread(void*) 
    #24 0x7fbc0ac5a1c2 in start_thread 
```
```
Direct leak of 96 byte(s) in 1 object(s) allocated from:
    #0 0x19fd390 in operator new(unsigned long) 
    #1 0x374618f in doris::ParquetScanner::open_next_reader() 
    #2 0x374532f in doris::ParquetScanner::get_next(doris::Tuple*, doris::MemPool*, bool*) 
    #3 0x36e229d in doris::BrokerScanNode::scanner_scan(doris::TBrokerScanRange const&, std::vector<doris::ExprContext*, std::allocator<doris::ExprContext*> > const&, std::vector<doris::ExprContext*, std::allocator<doris::ExprContext*> > const&, doris::ScannerCounter*) 
    #4 0x36e3135 in doris::BrokerScanNode::scanner_worker(int, int) ../src/exec/broker_scan_node.cpp:371
    #5 0x36e61e0 in void std::__invoke_impl<void, void (doris::BrokerScanNode::*)(int, int), doris::BrokerScanNode*, int, unsigned long>(std::__invoke_memfun_deref, void (doris::BrokerScanNode::*&&)(int, int), doris::BrokerScanNode*&&, int&&, unsigned long&&) 
    #6 0x36e5772 in std::__invoke_result<void (doris::BrokerScanNode::*)(int, int), doris::BrokerScanNode*, int, unsigned long>::type std::__invoke<void (doris::BrokerScanNode::*)(int, int), doris::BrokerScanNode*, int, unsigned long>(void (doris::BrokerScanNode::*&&)(int, int), doris::BrokerScanNode*&&, int&&, unsigned long&&) 
    #7 0x36e676a in decltype (__invoke((_S_declval<0ul>)(), (_S_declval<1ul>)(), (_S_declval<2ul>)(), (_S_declval<3ul>)())) std::thread::_Invoker<std::tuple<void (doris::BrokerScanNode::*)(int, int), doris::BrokerScanNode*, int, unsigned long> >::_M_invoke<0ul, 1ul, 2ul, 3ul>(std::_Index_tuple<0ul, 1ul, 2ul, 3ul>) 
    #8 0x36e669d in std::thread::_Invoker<std::tuple<void (doris::BrokerScanNode::*)(int, int), doris::BrokerScanNode*, int, unsigned long> >::operator()() 
    #9 0x36e660d in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (doris::BrokerScanNode::*)(int, int), doris::BrokerScanNode*, int, unsigned long> > >::_M_run() 
    #10 0x4d6e66e in execute_native_thread_routine 
```
```
Direct leak of 480 byte(s) in 3 object(s) allocated from:
    #0 0x19fd6d0 in operator new(unsigned long, std::nothrow_t const&) 
    #1 0x383c88e in doris::RowBlockAllocator::allocate(doris::RowBlock**, unsigned long, bool) 
    #2 0x3842cf9 in doris::SchemaChangeWithSorting::process(std::shared_ptr<doris::RowsetReader>, doris::RowsetWriter*, std::shared_ptr<doris::Tablet>, std::shared_ptr<doris::Tablet>) 
    #3 0x3851a1f in doris::SchemaChangeHandler::_convert_historical_rowsets(doris::SchemaChangeHandler::SchemaChangeParams const&) 
    #4 0x384a1b2 in doris::SchemaChangeHandler::_do_process_alter_tablet_v2(doris::TAlterTabletReqV2 const&) 
    #5 0x38478bb in doris::SchemaChangeHandler::process_alter_tablet_v2(doris::TAlterTabletReqV2 const&) 
    #6 0x392303b in doris::EngineAlterTabletTask::execute() 
    #7 0x1e693df in doris::StorageEngine::execute_task(doris::EngineTask*) 
    #8 0x3312e36 in doris::TaskWorkerPool::_alter_tablet(doris::TAgentTaskRequest const&, long, doris::TTaskType::type, doris::TFinishTaskRequest*) 
    #9 0x331254d in doris::TaskWorkerPool::_alter_tablet_worker_thread_callback() 
    #10 0x33380e1 in void std::__invoke_impl<void, void (doris::TaskWorkerPool::*&)(), doris::TaskWorkerPool*&>(std::__invoke_memfun_deref, void (doris::TaskWorkerPool::*&)(), doris::TaskWorkerPool*&) 
    #11 0x3337c30 in std::__invoke_result<void (doris::TaskWorkerPool::*&)(), doris::TaskWorkerPool*&>::type std::__invoke<void (doris::TaskWorkerPool::*&)(), doris::TaskWorkerPool*&>(void (doris::TaskWorkerPool::*&)(), doris::TaskWorkerPool*&) 
    #12 0x33372f6 in _ZNSt12_Bind_resultIvFMN5doris14TaskWorkerPoolEFvvEPS1_EE6__callIvJEJLm0EEEENSt9enable_ifIXtlSt7is_voidIT_EEEvE4typeEOSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE 
    #13 0x3335db6 in void std::_Bind_result<void, void (doris::TaskWorkerPool::*(doris::TaskWorkerPool*))()>::operator()<>() 
    #14 0x333462b in std::_Function_handler<void (), std::_Bind_result<void, void (doris::TaskWorkerPool::*(doris::TaskWorkerPool*))()> >::_M_invoke(std::_Any_data const&) 
    #15 0x27345a3 in std::function<void ()>::operator()() const 
    #16 0x2aeb219 in doris::FunctionRunnable::run() 
    #17 0x2ae819f in doris::ThreadPool::dispatch_thread()
    #18 0x2b01b61 in void std::__invoke_impl<void, void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(std::__invoke_memfun_deref, void (doris::ThreadPool::*&)(), doris::ThreadPool*&) 
    #19 0x2b00526 in std::__invoke_result<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>::type std::__invoke<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(void (doris::ThreadPool::*&)(), doris::ThreadPool*&) 
    #20 0x2afea84 in void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) 
    #21 0x2afc55e in void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::operator()<, void>() 
    #22 0x2af7cbd in std::_Function_handler<void (), std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()> >::_M_invoke(std::_Any_data const&) 
    #23 0x27345a3 in std::function<void ()>::operator()() const 
    #24 0x2acc9ca in doris::Thread::supervise_thread(void*) 
    #25 0x7fd6ee4c31c2 in start_thread 
```
**Expected behavior**

BE should't crash and shouldn't have any memory leak

**Additional context**
Later I will make a PR fix all DCHECK Fail In debug Mode

### PR DESCRIPTION
**Pull request by stdpain**
## Proposed changes
will close #5466 
will close #5476 

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue 
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If these changes need document changes, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/analytic_eval_node.cpp [hunk 1]
```diff
@@ -753,7 +753,8 @@ Status AnalyticEvalNode::get_next_output_batch(RuntimeState* state, RowBatch* ou
         // CopyRow works as expected: input_batch tuples form a prefix of output_batch
         // tuples.
         TupleRow* dest = output_batch->get_row(output_batch->add_row());
-        input_batch.copy_row(input_batch.get_row(i), dest);
+        input_batch.get_row(i)->deep_copy(dest, child(0)->row_desc().tuple_descriptors(),
```
Comment (morningman) [inline be/src/exec/analytic_eval_node.cpp:756]: What is different between copy_row and deep_copy?
modify the comment explain why

File: be/src/exec/olap_utils.h [hunk 2]
```diff
@@ -207,6 +207,9 @@ inline SQLFilterOp to_olap_filter_type(TExprOpcode::type type, bool opposite) {
     case TExprOpcode::NE:
         return opposite ? FILTER_IN : FILTER_NOT_IN;
 
+    case TExprOpcode::EQ_FOR_NULL:
```
Comment (morningman) [inline be/src/exec/olap_utils.h:210]: Can the storage engine handle `EQ_FOR_NULL` correctly with `FILTER_IN`?

File: be/src/exec/partitioned_aggregation_node.cc [hunk 3]
```diff
@@ -345,9 +345,12 @@ Status PartitionedAggregationNode::open(RuntimeState* state) {
...
 Status PartitionedAggregationNode::get_next(RuntimeState* state, RowBatch* row_batch, bool* eos) {
-    int first_row_idx = row_batch->num_rows();
-    RETURN_IF_ERROR(GetNextInternal(state, row_batch, eos));
-    RETURN_IF_ERROR(HandleOutputStrings(row_batch, first_row_idx));
+    DCHECK_EQ(row_batch->num_rows(), 0);
+    RowBatch batch(row_batch->row_desc(), row_batch->capacity(), _mem_tracker.get());
+    int first_row_idx = batch.num_rows();
+    RETURN_IF_ERROR(GetNextInternal(state, &batch, eos));
+    RETURN_IF_ERROR(HandleOutputStrings(&batch, first_row_idx));
+    batch.deep_copy_to(row_batch);
```
Comment (morningman) [inline be/src/exec/partitioned_aggregation_node.cc:353]: Add comment to explain why we need deep copy here

File: be/src/olap/aggregate_func.h [hunk 4]
```diff
@@ -461,10 +461,9 @@ struct AggregateFuncTraits<OLAP_FIELD_AGGREGATION_HLL_UNION, OLAP_FIELD_TYPE_HLL
         // we use zero size represent this slice is a agg object
         dst_slice->size = 0;
         auto* hll = new HyperLogLog(*src_slice);
+        
         dst_slice->data = reinterpret_cast<char*>(hll);
 
-        mem_pool->mem_tracker()->Consume(sizeof(HyperLogLog));
```
Comment (morningman) [inline be/src/olap/aggregate_func.h:466]: Why remove this?

File: be/src/runtime/tablets_channel.cpp [hunk 5]
```diff
@@ -260,7 +260,6 @@ Status TabletsChannel::cancel() {
     for (auto& it : _tablet_writers) {
         it.second->cancel();
     }
-    DCHECK_EQ(_mem_tracker->consumption(), 0);
```
Comment (morningman) [inline be/src/runtime/tablets_channel.cpp:263]: Why remove this?

File: be/src/exec/olap_utils.h (hunk 2 above)
Comment (stdpain) [reply be/src/exec/olap_utils.h:210]: It's OK. 
case 1:
```sql
select * from t1 where k1 <=> 1;
```
It's equals to EQ

case 2:
```
select * from t1 where k1 <=> null;
```
type  of k1 will not equals to LITERAL_NULL,  so storage engine will not use this conjunct

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**